### ISSUE TITLE
Getting internal server error after running for some months hPutBuf: resource vanished (Broken pipe)

### ISSUE THREAD

**Issue by daniellandau** [yesodweb/yesod#1244]
I have a web service written with Yesod and after some months running it starts spewing out internal server errors with the message

    &lt;stderr&gt;: hPutBuf: resource vanished (Broken pipe)

I am using `runStdErrLoggingT` and running as a Systemd service (which catches stdout and stderr to the journal). `systemctl status [service]` gives

    Warning: Journal has been rotated since unit was started. Log output is incomplete or unavailable.

Any help appreciated, this issue only comes after running some months so it's quite hard for me to debug.

**Comment by snoyberg** [comment]
I don't think this issue is Yesod-specific. I don't have experience running Haskell apps as a systemd service. Is there some prior documentation somewhere on the possibility of file descriptors being closed?

**Comment by daniellandau** [comment]
There is some documentation at https://www.freedesktop.org/software/systemd/man/systemd.exec.html#StandardOutput=. Other services still log fine to the journal.

This might not be Yesod-specific true, I'll be happy to get any pointers towards a solution.

**Comment by snoyberg** [comment]
Perhaps asking on Stack Overflow or the haskell-cafe will get more
responses. I'm sure there are others who have significant experience with
systemd.
